scripts/insert_yml.py

```
@@ -0,0 +1,120 @@
+import json
+import os
+from pathlib import Path
+
+
+def insert_to_ipynb(notebook_path: Path, n: int):
+    try:
+        with open(notebook_path, encoding="utf-8") as f:
+            nb = json.load(f)
+    except json.JSONDecodeError as e:
+        print(f"Error loading {notebook_path}: {e}")
+        return None
+
+    if nb is None:
+        print(f"Notebook {notebook_path} is empty or malformed.")
+        return None
+
+    n_cells_checked = 0
+    for cell in nb["cells"]:
+        if cell["cell_type"] == "markdown":
+            index_start = get_index_in_cell(
+                "<!-- START ENV-SETUP -->\n", cell["source"]
+            )
+            index_end = get_index_in_cell("<!-- END ENV-SETUP -->\n", cell["source"])
+
+            if index_start is not None and index_end is not None:
+                del cell["source"][index_start + 1 : index_end]
+                cell["source"][index_start + 1 : index_start] = [
+                    get_env_setup_str(notebook_path, md_env_setup)
+                ]
+                return nb
+
+        n_cells_checked += 1
+
+        if n_cells_checked >= n:
+            return nb
+
+    return nb
+
+
+def insert_to_md(md_path: Path, n: int):
+    try:
+        with open(md_path, encoding="utf-8") as f:
+            content = f.readlines()
+    except Exception as e:
```

Try not to catch `Exception` but more specific exceptions.
Also, should this not reraise the exception instead of printing an error?

---

Raised the error and caught `FileNotFoundError`.